tests/Parallel_AttributesTest.cpp

```diff
@@ -145,8 +157,115 @@ void Parallel_AttributesTest::testDataAttributes()
     CPPUNIT_ASSERT(strcmp(string_read, string_attr) == 0);
     CPPUNIT_ASSERT(strcmp(string_read4, string_attr4) == 0);
 
+    dataCollector->close();
     dataCollector->finalize();
+    delete dataCollector;
+}
+
+// Delete old info, read new info and check validity (reduce noise in test code)
+#define READ_ATTR_META(name, group) delete info; info = NULL;                           \
```

Comment:
Macros make it easier and shorter (name not duplicated). Especially as a failure of the test assert would indicate the correct line